designs/credentialManagement/profileFileSupport.md

```diff
@@ -0,0 +1,54 @@
+# Profile File Support
+
+## Abstract
+
+This document describes the process of supporting the AWS Profile files in the [Credential Management](credentialManagement.md) system
+
+## Shared Credentials File(s)
+
+AWS SDKs and tools are capable of sharing credentials by storing them in a common location. Also known as shared credential files, the **config** and 
+**credentials** files are detailed at [CLI documentation][CliConfigDocs]. We try to comply as close as possible with the CLI credential resolution behavior, 
+but only the keys related to credential management are supported in the toolkit. 
+
+Supported keys:
+* `aws_access_key_id`
+* `aws_secret_access_key`
+* `aws_session_token`
+* `source_profile`
+* `external_id`
+* `role_session_name`
+* `mfa_serial`
+* `credential_process`
+* `sso_*` - See [SSO Document](ssoSupport.md)
+
+### File Parsing
+Parsing and merging of the **config** and **credentials** files is out of scope of this document and is handled by the Java SDK. Please see [ProfileFile](https://github.com/aws/aws-sdk-java-v2/blob/master/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFile.java).
+
+If there are any issues parsing either of the files, the entire parsing job MUST be treated as a failure and the error shown to the user and no state changes 
+are made to the `CredentialManager`.
+
+### File Locating Resolution
+Locating of the profile files is handled by the Java SDK. Please see [ProfileFileLocation](https://github.com/aws/aws-sdk-java-v2/blob/master/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileLocation.java).
+
+If file location logic returns an invalid path, the files must be treated as not existing.
+
+### Profile File Modification Detection
+
+An Application Service (`ProfileWatcher`) is registered by the toolkit. The service instructs the IDE to notify it when the parent directory of the `config` 
+and/or `credentials` files are modified. We monitor the parent in folder in case the files do not exist yet.
```

Comment:
"the parent in folder" -> "the parent of the folder"?

Reply:
fixed